sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/UnsupportedOperationsSuite.scala

```diff
@@ -413,36 +414,103 @@ class UnsupportedOperationsSuite extends SparkFunSuite {
     batchStreamSupported = false,
     streamBatchSupported = false)
 
-  // Left outer joins: *-stream not allowed
+  // Left outer joins: *-stream not allowed with default condition
   testBinaryOperationInStreamingPlan(
     "left outer join",
     _.join(_, joinType = LeftOuter),
-    streamStreamSupported = false,
     batchStreamSupported = false,
-    expectedMsg = "left outer/semi/anti joins")
+    streamStreamSupported = false,
+    expectedMsg = "outer join")
+
+  // Left outer joins: stream-stream allowed with join on watermark attribute
+  assertSupportedInStreamingPlan(
+    s"left outer join with stream-stream relations and join on watermark attribute key",
+    streamRelation.join(streamRelation, joinType = LeftOuter,
+      condition = Some(attributeWithWatermark === attributeWithWatermark)),
+    OutputMode.Append())
+
+  // Left outer joins: stream-stream allowed with range condition yielding state value watermark
+  assertSupportedInStreamingPlan(
+    s"left outer join with stream-stream relations and state value watermark", {
+      val firstRelationWithWatermark = new TestStreamingRelation(attributeWithWatermark)
+      val secondAttribute = AttributeReference("b", IntegerType)().withMetadata(watermarkMetadata)
+      firstRelationWithWatermark.join(
+        new TestStreamingRelation(secondAttribute),
+        joinType = LeftOuter,
+        condition = Some(secondAttribute < attributeWithWatermark + 10 &&
```

Comment:
i think conditions on both sides should not be required. For LeftOuter, we only need `leftAttribute > rightAttributeWithWatermark + c`

Reply:
also... keep the left attribute on left side of the condition. maybe its just me, its just easier to read and reason.